iModelCore/BeSQLite/BeSQLite.cpp

```diff
@@ -2786,6 +2786,36 @@ DbResult Db::DetachDb(Utf8CP alias) const
     return rc;
     }
 
+/*---------------------------------------------------------------------------------**//**
+*
++---------------+---------------+---------------+---------------+---------------+------*/
+std::vector<AttachFileInfo> Db::GetAttachedDbs() const {
+    if (!IsDbOpen())
+        return {};
+
+    std::vector<AttachFileInfo> result;
+    Statement stmt;
+    stmt.Prepare(*this, "PRAGMA database_list");
+    while (stmt.Step() == BE_SQLITE_ROW) {
+        AttachFileInfo info;
+        info.m_alias = stmt.GetValueText(1);
+        info.m_fileName = stmt.GetValueText(2);
+        if (info.m_alias.EqualsIAscii("main")) {
+            info.m_type = AttachFileTypes::Main;
+        } else if (info.m_alias.EqualsIAscii("schema_sync_db")){
+            info.m_type = AttachFileTypes::SchemaSync;
+        } else if (info.m_alias.EqualsIAscii("ecchange")){
+            info.m_type = AttachFileTypes::ECChangeCache;
+        } else if (info.m_alias.EqualsIAscii("temp")){

```

Comment:
It looks like these specific aliases have special meaning. If I'm not misunderstanding, there should probably be documentation and code on the TS side forbidding their use as aliases in the calls to `attachDb`. It seems like if someone calls `attachDb("somefile", "main")`, the call will succeed, but that it will cause problems. I could be wrong about all of this, since I don't fully understand the native logic.

Reply:
No the call will fail. It down to sqlite which should fail the attach and error would be returned. This enum was for query reader api which must attach each of these over to a separate sqlite connection held by separate thread. I like to avoid attaching main/temp/sync_db to concurrent query secondary connection.

Reply:
I'm saying that these special aliases should be documented in the two `attachDb` functions in itwinjs-core. Otherwise people will get an exception thrown at run-time and not understand why.

Reply:
ah ok will do

Reply:
so i added docs to itwin.js side